### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by Chris-Costa**
-Changes to general layout of app adding guards for two tabs when user is not auth.
-Added to existing exercise list component to create new workouts.  This data is sent to new component workout list and is displayed on its own page. 
-Added a profile editing component for first and last name as well as avatar for gender of user.  
-Basic blog component and form for new posts.
-Other angular material styling updates

### CODE REVIEW

File: cvFit/src/app/exercises/workoutList/dataTransfer.service.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { Injectable } from "@angular/core";
+
+@Injectable()
+export class TransferService {
```
Comment (swhook52) [inline cvFit/src/app/exercises/workoutList/dataTransfer.service.ts:4]: This is another one where I'm not sure what it's doing. This looks like a service to share data with other components? The variables and actions could be better with some renames. Then maybe when we understand what it's doing we might not even have a need for the service at all.

**Review by swhook52 — CHANGES_REQUESTED** [review]
I think i'm only through about half but I ran out of time.

File: cvFit/src/app/exercises/exercise-list.component.ts [hunk 2]
```diff
@@ -37,11 +54,67 @@ export class ExerciseListComponent implements OnInit {
...
-}
\ No newline at end of file
+    create(): void{
+        this.createNew = !this.createNew;
+    }
+
+    tempArray: string[] = [];
+    arrayToSend: number[] = [];
+    show: boolean = false;
+    bool: boolean = true;
```
[the remark this answers is not in the document]
Comment (Chris-Costa) [reply cvFit/src/app/exercises/exercise-list.component.ts:70]: renamed these and added some comments to this file.  also pulled all variables to the top

File: cvFit/src/app/blog/blog.component.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    openDialog() {
+        this.dialog.open(PostComponent, {
+          width: '500px',
+        });
+    }
+
+    arr: number[] = []
+    disableLikedPostsButton: boolean = false
+    //hold liked blog posts
+    filteredList: IBlogPost[] = [
```
[the remark this answers is not in the document]
Comment (Chris-Costa) [reply cvFit/src/app/blog/blog.component.ts:52]: that makes sense.  I was thinking about this with the workout list as well.  Something like that should also reference the user id

File: cvFit/src/app/blog/blog.component.html [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  </div>
+
+  </mat-sidenav>
+ 
+  <!--HTML for all posts-->
+  <mat-sidenav-content *ngIf="allFlag">
+    <div class="flex-container1">
+      <mat-card class="blog-card1" *ngFor='let blogPost of blogPosts'>
+        <mat-card-title-group>
+          <mat-card-title><a id="exerName" [routerLink]="['/discussion', blogPost.blogId]">
```
[the remark this answers is not in the document]
Comment (Chris-Costa) [reply cvFit/src/app/blog/blog.component.html:21]: I'm confused about this comment.  Currently this routerLink works as intended and will expand the details of each unique blog post.  I also have this code in the exercise list -> exercise detail components.

File: cvFit/src/app/blog/discussion/discussion.component.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+
+@Component({
+    selector: 'app-discussion',
+    templateUrl: './discussion.component.html',
+    styleUrls: ['./discussion.component.css']
+})
+export class DiscussionComponent{
+    constructor(private route: ActivatedRoute, private router: Router, private blogService: BlogService, public auth: AuthService) { }
+    holdId: number;
```
[the remark this answers is not in the document]
Comment (Chris-Costa) [reply cvFit/src/app/blog/discussion/discussion.component.ts:18]: renamed to commentsBlogId (still feel like this name is a bit confusing), using this variable to give an id to the comments.  In the html if this comment id and id of the blogPost match the comment belongs to that blog post and should be displayed when details of the post are expanded.

File: cvFit/src/app/app.component.ts [hunk 6]
```diff
@@ -1,10 +1,26 @@
...
+  constructor(public auth:AuthService, public dialog: MatDialog, private router: Router) {}
+  
+  openDialog() {
+    this.dialog.open(ProfileComponent, {
+      width: '500px',
+    });
+  }
+  logout(){
+    this.router.navigate(['welcome']);
+    this.auth.logout().subscribe(() => {
```
[the remark this answers is not in the document]
Comment (Chris-Costa) [reply cvFit/src/app/app.component.ts:22]: used a take(1)

File: cvFit/src/app/blog/blog.component.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (Chris-Costa) [reply cvFit/src/app/blog/blog.component.html:21]: I ended up changing this code anyways.  The routerLink portion has been replaced with a (click) event that receives the Id

File: cvFit/src/app/blog/discussion/discussion.component.ts [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            this.holdId = id;
+        }
+    }
+    //routing with code
+    //go from detail expansion page back to exercise list
+    onBack(): void {
+        this.router.navigate(['/blog']);
+    } 
+    getPost(id: number): void {
+        this.blogService.getBlogPost(id).subscribe({
```
[the remark this answers is not in the document]
Comment (Chris-Costa) [reply cvFit/src/app/blog/discussion/discussion.component.ts:40]: using async here

**Comment by Chris-Costa** [comment]
Changes made based on the comments.  Still need to work through getting a complete list of liked blog posts(currently hardcoded to produce one post).  Will be working to fix similar observable issues that are present in other files that aren't a part of the previous review (exercise list and exercise details).  Let me about the dataTransfer service and best ways to move forward with that code.

**Review by swhook52 — APPROVED** [review]

**Pull request merged**